Source/WebCore/svg/graphics/SVGImage.cpp

```diff
@@ -528,16 +536,14 @@ bool isInSVGImage(const Element* element)
     return page->chrome().client().isSVGImageChromeClient();
 }
 
-bool SVGImage::isDataDecodable(const Settings& settings, std::span<const uint8_t> data)
+RefPtr<SVGImage> SVGImage::createFromData(std::span<const uint8_t> data)
```

Comment:
I think we should have both functions: `isDataDecodable()` and `createFromData()`. The caller of `isDataDecodable()` does not need to create an `SVGImage` and sets the data to it to know the value of `isDataDecodable()`. The document result of the parser should be enough.

Maybe you can create a function to have a the common code:
```
static RefPtr<Document> createDocumentFromSVGImageData()
{
    ...
}
```

Reply:
I left this improvement for the future as I'm not sure how to completely bypass `DocumentLoader` and such.